aztokenprovider/retriever_wi.go

```diff
@@ -0,0 +1,72 @@
+package aztokenprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+)
+
+type workloadIdentityTokenRetriever struct {
+	tenantId   string
+	clientId   string
+	credential azcore.TokenCredential
+}
+
+func getWorkloadIdentityTokenRetriever(settings *azsettings.AzureSettings, credentials *azcredentials.AzureWorkloadIdentityCredentials) TokenRetriever {
+	var tenantId, clientId string
+	// TODO: See https://azure.github.io/azure-workload-identity/docs/faq.html#how-to-federate-multiple-identities-with-a-kubernetes-service-account
```

Comment:
Should we add support for this immediately?

Reply:
Added